public/swagger/swagger.json

```diff
@@ -0,0 +1,997 @@
+            "description": "The id of the rule",
+            "type": "string",
+            "required": true
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "rule state sorted latest first",
+            "schema": {
+              "$ref": "#/definitions/RollupRuleHistory"
+            }
+          },
+          "404": {
+            "description": "no such namespace or no such rule",
+            "schema": {
+              "$ref": "#/definitions/ApiResponse"
+            }
+          },
+          "500": {
+            "description": "Something went horribly wrong",
+            "schema": {
+              "$ref": "#/definitions/ApiResponse"
+            }
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "Namespace": {
+      "type": "object",
+      "properties": {
+        "name": {
+          "type": "string"
+        }
+      }
+    },
+    "Namespaces": {
+      "type": "array",
+      "items": {
+        "$ref": "#/definitions/Namespace"
+      }
+    },
+    "RuleSet": {
+      "type": "object",
+      "properties": {
+        "namespace": {
+          "type": "string"
+        },
+        "version": {
+          "type": "integer"
+        },
+        "cutoverTime": {
+          "type": "string",
+          "format": "date-time"
+        },
+        "mappingRules": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/MappingRule"
+          }
+        },
+        "rollupRules": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/RollupRule"
+          }
+        }
+      }
+    },
+    "MappingRule": {
+      "type": "object",
+      "properties": {
+        "id": {
+          "type": "string",
+          "format": "uuid"
+        },
+        "name": {
+          "type": "string"
+        },
+        "filters": {
+          "type": "string"
+        },
+        "policies": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "cutoverNanos": {
+          "type": "integer",
+          "description": "unixNanos"
+        },
+        "lastUpdateBy": {
+          "type": "string"
+        },
+        "lastUpdatedNanos": {
+          "type": "integer",
+          "description": "unixNanos"
+        }
+      }
+    },
+    "RollupRule": {
+      "type": "object",
+      "properties": {
+        "id": {
+          "type": "string",
+          "format": "uuid"
+        },
+        "name": {
+          "type": "string"
+        },
+        "filters": {
+          "type": "string"
+        },
+        "targets": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/RollupTarget"
+          }
+        },
+        "cutoverNanos": {
+          "type": "integer",
+          "description": "unixNanos"
+        },
+        "lastUpdateBy": {
+          "type": "string"
+        },
+        "lastUpdatedNanos": {
```

Comment:
Is it important to have the timestamps in nanos instead of milliseconds? In other words, do we need this much precision?

Some clients don't have great support for nanoseconds (eg Javascript)